Source/items.cpp

```diff
@@ -4453,55 +4453,46 @@ void SpawnBoy(int lvl)
 
 BOOL HealerItemOk(int i)
 {
-	int v1;         // ecx
-	BOOLEAN result; // eax
-	int v3;         // esi
+	BOOL result;
 
-	v1 = i;
 	result = 0;
-	if (AllItemsList[v1].itype)
+	if (AllItemsList[i].itype != ITYPE_MISC)
 		return 0;
-	v3 = AllItemsList[v1].iMiscId;
-	if (v3 == IMISC_SCROLL && AllItemsList[v1].iSpell == SPL_HEAL)
+
+	if (AllItemsList[i].iMiscId == IMISC_SCROLL && AllItemsList[i].iSpell == SPL_HEAL)
 		result = 1;

```

Comment:
Since it's `BOOL`, should be `TRUE`/`FALSE` right?

Reply:
yes :)